presto-main/src/main/java/com/facebook/presto/sql/gen/PageFunctionCompiler.java

```diff
@@ -839,13 +866,8 @@ public BytecodeNode visitLambda(LambdaDefinitionExpression lambda, Scope context
         public BytecodeNode visitVariableReference(VariableReferenceExpression reference, Scope context)
         {
             CommonSubExpressionFields fields = variableMap.get(reference);
-            IfStatement ifStatement = new IfStatement()
-                    .condition(thisVariable.getField(fields.evaluatedField))
-                    .ifFalse(new BytecodeBlock()
-                            .append(thisVariable.invoke(fields.methodName, void.class, context.getVariable("properties"), context.getVariable("page"), context.getVariable("position"))));
             return new BytecodeBlock()
-                    .append(ifStatement)
-                    .append(thisVariable.getField(fields.resultField))
+                    .append(thisVariable.invoke(fields.methodName, fields.resultType, context.getVariable("properties"), context.getVariable("page"), context.getVariable("position")))
```

Comment:
When you are at it - is it feasible to make all these also fields instead of passing so many arguments?

Reply:
What do you mean? They are function parameters in `evaluate` as well, and these functions are called from `evaluate`. I think it makes more sense to keep them consistent.

Reply:
I guess so. It's just that the whole param passing is unnecessary. Saves a few bytes in the method size too :) Hopefully the JIT will remove all that at least from performance point of view, it should be ok